### PR TITLE
SONAR-9165 Allow to change project visibility on its permissions page

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/permissions/project/components/PageHeader.js [hunk 1]
```diff
@@ -17,47 +17,51 @@
...
-import { loadHolders } from '../store/actions';
-import { isPermissionsAppLoading } from '../../../../store/rootReducer';
 
-class PageHeader extends React.PureComponent {
-  static propTypes = {
-    project: React.PropTypes.object.isRequired,
-    loadHolders: React.PropTypes.func.isRequired,
-    loading: React.PropTypes.bool
-  };
+type Props = {|
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/permissions/project/components/PageHeader.js]: MAJOR Parse error [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AParsingError)

File: server/sonar-web/src/main/js/apps/permissions/project/components/PublicProjectDisclaimer.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// @flow
+import React from 'react';
+import Modal from 'react-modal';
+import { translate, translateWithParameters } from '../../../../helpers/l10n';
+
+type Props = {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/permissions/project/components/PublicProjectDisclaimer.js]: MAJOR Parse error [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AParsingError)

File: server/sonar-web/src/main/js/apps/permissions/project/components/App.js [hunk 3]
```diff
@@ -17,39 +17,357 @@
...
-import { getComponent, getCurrentUser } from '../../../../store/rootReducer';
+import * as api from '../../../../api/permissions';
 import '../../styles.css';
 
 // TODO helmet
 
-class App extends React.PureComponent {
-  static propTypes = {
-    component: React.PropTypes.object
+export type Props = {|
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/permissions/project/components/App.js]: MAJOR Parse error [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AParsingError)

File: server/sonar-web/src/main/js/apps/permissions/project/components/VisibilitySelector.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// @flow
+import React from 'react';
+import classNames from 'classnames';
+import { translate } from '../../../../helpers/l10n';
+
+type Props = {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/permissions/project/components/VisibilitySelector.js]: MAJOR Parse error [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AParsingError)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by stas-vilchik** [comment]
superseded by #2002

**Pull request closed without merging**